src/vpu/encoder_h264.c

```diff
@@ -256,6 +273,16 @@ GstCaps* gst_imx_vpu_encoder_h264_get_output_caps(GstImxVpuEncoderBase *vpu_enco
 gboolean gst_imx_vpu_encoder_h264_set_frame_enc_params(GstImxVpuEncoderBase *vpu_encoder_base, ImxVpuEncParams *enc_params)
 {
 	GstImxVpuEncoderH264 *vpu_encoder_h264 = GST_IMX_VPU_ENCODER_H264(vpu_encoder_base);
+	ImxVpuFramebuffer *input_framebuffer = &vpu_encoder_base->input_framebuffer;
+
+
+	if (input_framebuffer->cb_offset == 0 && input_framebuffer->cr_offset == 0)
```

Comment:
I'd rather look up the open_params from vpu_encoder_base and check if color_format == IMX_VPU_COLOR_FORMAT_YUV400